app/src/main/java/com/epfl/drawyourpath/userProfile/UserModel.kt

```diff
@@ -0,0 +1,343 @@
+                this.distanceGoal=distanceGoal
+            }
+            it
+        }
+    }
+
+    /**
+     * Get the daily activity time goal of the user
+     * @return daily activity time goal of the user
+     */
+    fun getActivityTime(): Double{
+        return activityTimeGoal
+    }
+
+    /**
+     * Use this function to modify the daily activity time goal of the user
+     * @param time new daily activity time goal
+     */
+    fun setActivityTimeGoal(activityTimeGoal: Double): CompletableFuture<Boolean>{
+        checkActivityTimeGoal(activityTimeGoal)
+        return database.setActivityTimeGoal(userId, activityTimeGoal).thenApply {
+            if(it){
+                this.activityTimeGoal=activityTimeGoal
+            }
+            it
+        }
+    }
+
+    /**
+     * Get the daily number of paths goal of the user
+     * @return daily number of paths goal of the user
+     */
+    fun getNumberOfPathsGoal(): Int{
+        return nbOfPathsGoal
+    }
+
+    /**
+     * Use this function to modify the daily number of paths goal of the user
+     * @param nbOfPaths new daily number of paths goal
+     */
+    fun setNumberOfPathsGoal(nbOfPathsGoal: Int): CompletableFuture<Boolean>{
+        checkNbOfPathsGoal(nbOfPathsGoal)
+        return database.setNbOfPathsGoal(userId, nbOfPathsGoal).thenApply {
+            if(it){
+                this.nbOfPathsGoal=nbOfPathsGoal
+            }
+            it
+        }
+    }
+
+    /**
+     * Get the age of the user
+     * @return the age of the user
+     */
+    fun getAge(): Int{
+        return ChronoUnit.YEARS.between(dateOfBirth, LocalDate.now()).toInt()
+    }
+
+    /**
+     * This function will remove the user with username to the friend list
+     * @param username of the user that we want to remove
+     */
+    fun removeFriend(username: String) {
+        if(!friendsList.contains(username)) {
+            throw java.lang.Error("This user is not in the friend list !")
+        }
+        friendsList.remove(username)
+        //TODO: Update the friend list in the database
+    }
+
+    /**
+     * This function will add the user with username to the friend list. To be added the user must be present in the database.
+     * @param username of the user that we want to add to the friend list
+     */
+    fun addFriend(username: String) {
+        //TODO:this function will be implemented during a next task when the database will be cleaned
+        //TODO: Update the friend list in the database
+    }
+
+    /**
+     * This function will return the friend list of a user
+     * @return the friend list of the user
+     */
+    fun getFriendList():Map<String, String> {
+        return this.friendsList
+    }
+
+}
+
+/**
+ * Helper function to check if the firstname format is correct and throw directly an error if it is incorrect
+ * @param name to be check
+ * @throw an error if the format is not correct
+ */
+private fun checkFirstnameFormat(name: String){
```

Comment:
`private fun checkFirstnameFormat(name: String)` and `private fun checkSurnameFormat(name: String)` are still the same function and code is duplicated. We can just have a single `checkNameFormat()` one I think.

Reply:
I think that if we just have a checkName function, we can't differentiate if the error come from the username or the first name

Reply:
Thank for your review. I m waiting your answer for merging.

Reply:
We could pass the checked property name as a string parameter and use it when throwing the exception.